src/main/java/frc/robot/note_tracking/NoteTrackingManager.java

```diff
@@ -358,40 +358,32 @@ public void addNoteToMap(Translation2d pose) {
     noteMap.add(new NoteMapElement(Timer.getFPGATimestamp() + NOTE_MAP_LIFETIME, pose, 10));
   }
 
-  private NoteMapElement updatedHealth(NoteMapElement oldElement) {
-    return new NoteMapElement(
-        oldElement.expiresAt(), oldElement.noteTranslation(), oldElement.health() - 1);
-  }
-
   private void updateMap() {
     List<Pose2d> visionNotes = getFilteredNotePoses();
 
-    var filteredNotesInBox =
-        noteMap.stream()
-            .filter(
-                element -> {
-                  return (RobotConfig.get().perfToggles().noteMapBoundingBox()
-                      && safeToTrack()
-                      && noteInView(element.noteTranslation()));
-                });
-
-    filteredNotesInBox.forEach(
+    noteMap.removeIf(
         element -> {
-          noteMap.remove(element);
+          return (element.expiresAt() < Timer.getFPGATimestamp());
         });
 
-    var updatedHealthNotes = filteredNotesInBox.map(this::updatedHealth).toList();
+    if (RobotConfig.get().perfToggles().noteMapBoundingBox() && safeToTrack()) {
 
-    updatedHealthNotes.stream()
-        .forEach(
-            element -> {
-              noteMap.add(element);
-            });
+      var filteredNotesInBox = noteMap.stream()
+          .filter(
+              element -> {
+                return (noteInView(element.noteTranslation()));
+              })
+          .toList();
+
+      for (NoteMapElement noteMapElement : filteredNotesInBox) {
+        noteMap.remove(noteMapElement);
+        if (noteMapElement.health() >= 1) {
```

Comment:
in this case we want it to be
```suggestion
        if (noteMapElement.health() > 1) {
```

this is for saying "if this note was about to hit `health == 0`, just remove it without actually updating it first"